src/data_readers/data_reader_hdf5.cpp

```diff
@@ -0,0 +1,237 @@
+        int xPerNode = dims[0]/xlines;
+        int yPerNode = dims[1]/ylines;
+        int zPerNode = dims[2]/zlines;
+        int cPerNode = dims[3]/channellines;
+       	//TODO: change this to be allocated elsewhere
+        short int data_out[xPerNode*yPerNode*zPerNode*cPerNode]; 
+        hsize_t dims_local[4];
+        hsize_t offset[4];
+        hsize_t count[4];
+        // local dimensions aka the dimensions of the slab we will read in
+        dims_local[0] = xPerNode;
+        dims_local[1] = yPerNode;
+        dims_local[2] = zPerNode;
+        dims_local[3] = cPerNode;
+        // necessary for the hdf5 lib
+        hid_t memspace = H5Screate_simple(4, dims_local, NULL);
+        int odd_offset;
+        if (xlines > 1) {
+            // this is theoretically the odd case but it has not been tested yet
+            // so everything with odd should be ignored
+            if((rank%num_io_parts) < int(dims[0]%xPerNode)) {
+                offset[0] = ((xPerNode+1)*(rank%num_io_parts));
+            } else {
+                // offset of this x dim for this rank;
+                // in most cases odd_offset will be 0
+                odd_offset = (xPerNode+1)*(dims[0]%xPerNode);
+                offset[0] = odd_offset +(xPerNode*((rank%num_io_parts)%xlines)); 
+            }
+        } else {
+           offset[0] = 0;
+        }
+        
+        if (ylines > 1) {
+            if((rank%num_io_parts) < int(dims[1]%yPerNode)) {
+                offset[1] = ((yPerNode+1)*(rank%num_io_parts));
+             } else {
+                // offset of the y dim for this rank
+                odd_offset = (yPerNode+1)*(dims[1]%yPerNode);
+                offset[1] = odd_offset + (yPerNode*((rank%num_io_parts)/ylines));
+             }   
+        } else {
+            offset[1] = 0;
+        }
+
+        if (zlines > 1) {
+            offset[2] = zPerNode*(rank%num_io_parts);
+        } else {
+            offset[2] = 0;
+        }
+        // I have channel dim splits 
+        // all combos arent really tested
+        //add the rest later
+        offset[3] = 0; 
+        count[0] = 1;
+        count[1] = 1;
+        count[2] = 1;
+        count[3] = 1;
+        // from an explanation of the hdf5 select_hyperslab:
+        // start -> a starting location for the hyperslab
+        // stride -> the number of elements to separate each element or block to be selected
+        // count -> the number of elemenets or blocks to select along each dimension
+        // block -> the size of the block selected from the dataspace 
+        //hsize_t status;
+        
+        //todo add error checking
+        H5Sselect_hyperslab(filespace, H5S_SELECT_SET, offset, NULL, count, dims_local);
+        H5Dread(h_data, H5T_NATIVE_SHORT, memspace, filespace, H5P_DEFAULT, data_out);
+    }
+
+    void hdf5_reader::load() {
+       
+        std::string dirpath = get_file_dir();    
+        std::vector<std::string> file_list = get_filenames(dirpath);
```

Comment:
You probably need to update this part as you're now using `glob` to grab a list of all sample files.